python/semantic_kernel/connectors/memory/weaviate/weaviate_collection.py

```diff
@@ -263,20 +389,16 @@ async def delete_collection(self, **kwargs) -> None:
         Args:
             **kwargs: Additional keyword arguments.
         """
-        async with self.async_client:
-            try:
-                await self.async_client.collections.delete(self.collection_name)
-            except Exception as ex:
-                raise MemoryConnectorException(f"Failed to delete collection: {ex}")
+        try:
+            await self.async_client.collections.delete(self.collection_name)
+        except Exception as ex:
+            raise MemoryConnectorException(f"Failed to delete collection: {ex}") from ex
 
     @override
     async def __aenter__(self) -> "WeaviateCollection":
         """Enter the context manager."""
-        if not await self.async_client.is_live():
-            try:
-                await self.async_client.connect()
-            except WeaviateConnectionError as exc:
-                raise MemoryConnectorConnectionException("Weaviate client cannot connect.") from exc
+        if not await self.async_client.is_ready():
+            await self.async_client.connect()
```

Comment:
I see you are connecting here but what if someone is not using the connector with an async context manager?